orbit/pkg/update/runner.go

```diff
@@ -140,11 +142,35 @@ func (r *Runner) HasLocalHash(target string) bool {
 	return ok
 }
 
+func randomizeDuration(min time.Duration, max time.Duration) (time.Duration, error) {
+	nBig, err := rand.Int(rand.Reader, big.NewInt(int64(max-min)))
+	if err != nil {
+		return 0, nil
+	}
+	return time.Duration(nBig.Int64()) + min, nil
+}
+
 // Execute begins a loop checking for updates.
 func (r *Runner) Execute() error {
 	log.Debug().Msg("start updater")
 
-	ticker := time.NewTicker(r.opt.CheckInterval)
+	// Randomize the initial interval so that all agents don't synchronize their updates
+	initialInterval := r.opt.CheckInterval
+	var randomizedInterval time.Duration
+	var err error
+	if initialInterval < 10*time.Minute {
+		// Developers use a shorter update interval (10s), so they need a faster first update check
+		randomizedInterval, err = randomizeDuration(0, initialInterval)
```

Comment:
I know I said OK to this but I missed the zero case.
I don't think we should allow a zero value.